File: Sine Up.py
from math import sin
from math import radians

# file = open("data\\sineup.dat")
file = open("D:\\JudgesData\\sineup.dat")

numbers = []

while True:
    data = file.readline().rstrip("\r\n")
    if data == '': break
    numbers.append([int(x) for x in data.split(" ")])

for number in numbers:
    D = number[0]
    A = number[1]
    B = number[2]
    C = 180 - A - B
    print "Circumcircle diameter = %d" % D
    print "Angles are %d, %d, and %d" % (A, B, C)
    print "Corresponding sides are %d, %d and %d" % (round(D * sin(radians(A)), 0), round(D * sin(radians(B)), 0), round(D * sin(radians(C)), 0))

File: src/main/java/me/timtim17/dev/pscsta/spring2015/java/SineUp.java
package me.timtim17.dev.pscsta.spring2015.java;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class SineUp {
    public static void main(String[] args) throws Exception {
        // Get input
        Scanner input;
        try {
            // try to get USB data
            input = new Scanner(new File("X:\\" + "SineUp.dat"));
        } catch (NullPointerException | FileNotFoundException e) {
            try {
                // no USB found, try to get student data
                input = new Scanner(SineUp.class.getClassLoader().getResourceAsStream("SineUp.dat"));
            } catch (NullPointerException er) {
                // neither could be found, throw an exception
                throw new Exception("Couldn't get data: check your file paths!");
            }
        }

        // Parse the data
        while (input.hasNext()) {
            String[] line = input.nextLine().split(" ");
            int[] nums = new int[3];
            for (int i = 0; i < line.length; i++) {
                nums[i] = Integer.valueOf(line[i]);
            }
            int diameter = nums[0];
            int angleADeg = nums[1], angleBDeg = nums[2], angleCDeg = 180 - angleADeg - angleBDeg;
            double angleARad = Math.toRadians(angleADeg), angleBRad = Math.toRadians(angleBDeg), angleCRad = Math.toRadians(angleCDeg);
            int sideA = (int) Math.round(diameter * Math.sin(angleARad)), sideB = (int) Math.round(diameter * Math.sin(angleBRad)), sideC = (int) Math.round(diameter * Math.sin(angleCRad));

            System.out.println("Circumcircle diameter = " + diameter);
            System.out.println("Angles are " + angleADeg + ", " + angleBDeg + " and " + angleCDeg);
            System.out.println("Corresponding sides are " + sideA + ", " + sideB + " and " + sideC);
        }

        // Close the file
        input.close();
    }
}
